liquid-compiler/src/filter.rs

```diff
@@ -1,101 +1,121 @@
-use std::fmt::{self, Debug};
-
-use liquid_error;
+use super::FilterArguments;
+use liquid_error::Result;
+use liquid_interpreter::Context;
 use liquid_value::Value;
+use std::fmt::Debug;
 
-/// Expected return type of a `Filter`.
-pub type FilterResult = Result<Value, liquid_error::Error>;
-
-/// A trait for creating custom tags. This is a simple type alias for a function.
-///
-/// This function will be called whenever the parser encounters a tag and returns
-/// a new [Renderable](trait.Renderable.html) based on its parameters. The received parameters
-/// specify the name of the tag, the argument [Tokens](lexer/enum.Token.html) passed to
-/// the tag and the global [`Language`](struct.Language.html).
-pub trait FilterValue: Send + Sync + FilterValueClone + Debug {
-    /// Filter `input` based on `arguments`.
-    fn filter(&self, input: &Value, arguments: &[Value]) -> FilterResult;
-}
 
-/// Support cloning of `Box<FilterValue>`.
-pub trait FilterValueClone {
-    /// Cloning of `dyn FilterValue`.
-    fn clone_box(&self) -> Box<FilterValue>;
+pub struct ParameterReflection {
+    name: &'static str,
```

Comment:
Do these fields need to be `pub`?

Reply:
Yes, they do. I completely forgot and as I still hadn't actually used the struct yet I didn't realize they were private.